### PR TITLE
[docs] Improve Next.js license installation guide

### PR DESCRIPTION
**Pull request by oliviertassinari**
Now that the [Next.js app folder is stable](https://nextjs.org/blog/next-13-4#nextjs-app-router), it makes more sense to document it.

~The 'use client' feels a bit like crap at this location, but I couldn't find any other API. I'm surprised Next.js doesn't have a global JavaScript injection point, typically for polyfills.~

Before: https://mui.com/x/introduction/licensing/#where-to-install-the-key
After: https://deploy-preview-8975--material-ui-x.netlify.app/x/introduction/licensing/#where-to-install-the-key

### CODE REVIEW

**Comment by oliviertassinari** [comment]
> The 'use client' feels a bit like crap at this location, but I couldn't find any other API. I'm surprised Next.js doesn't have a global JavaScript injection point, typically for polyfills.

Actually, creating an intermediate component should solve this.

**Comment by oliviertassinari** [comment]
@mnajdova @brijeshb42 one step forward for RSC.

Screenshot 2023-05-27 at 18 21 29

**Comment by oliviertassinari** [comment]
I'm merging since it was open close to a month ago.

**Review by richbustos — COMMENTED** [review]
LGTM

File: docs/data/introduction/licensing/licensing.md [hunk 1]
```diff
@@ -142,20 +143,70 @@ They include a dependency called `@mui/x-license-pro`, used to validate the lice
...
+
+```tsx
+import MuiXLicense from './MuiXLicense';
+
+export default function RootLayout(props: { children: React.ReactNode }) {
+  return (
+    <html lang="en">
+      <body>
+        {props.children}
+        <MuiXLicense />
```
Comment (cysp) [inline docs/data/introduction/licensing/licensing.md:200]: FYI: I had independently implemented something similar to this in my Next.js 13 appdir app to set the license key but without this no-op component, just relying on the side-effect of importing the licensing module to set the key. I'm not observing any issues in our deployed environment from this implementation approach.
Are there benefits to requiring this component in addition to the side-effect of setting the license key?

<details>
<summary>Nitty-gritty:</summary>

app/layout.tsx:

```tsx
import '../lib/mui-x/license'

// rest of layout
```

lib/mui-x/license.ts:

```tsx
'use client'

import { LicenseInfo } from '@mui/x-license-pro'

const licenseKey = process.env.NEXT_PUBLIC_MUI_X_LICENSE_KEY

if (licenseKey) {
  LicenseInfo.setLicenseKey(licenseKey)
}
```


</details>

**Pull request merged**